cmd/ci-operator/main_test.go

```diff
@@ -61,3 +76,204 @@ func TestSanitizeMessage(t *testing.T) {
+
+	// Verify without custom metadata
+	o := createOptions(tempDir, org, repo, extraOrg, extraRepo, prowJobID, namespace)
+	err = o.writeMetadataJSON()
+	if err != nil {
+		return fmt.Errorf("error while writing metadata JSON: %v", err)
+	}
+
+	err = verifyMetadataFields(metadataFile, org, repo, extraOrg, extraRepo, prowJobID, namespace, map[string]string{})
+
+	info, err := os.Stat(metadataFile)
+	if err != nil {
+		return fmt.Errorf("unable to stat metadata file: %v", err)
+	}
+
+	originalModTime := info.ModTime()
+
+	testArtifactDirectory := filepath.Join(tempDir, "test-artifact-directory")
+	if os.Mkdir(testArtifactDirectory, os.FileMode(0755)) != nil {
+		return fmt.Errorf("unable to create artifact directory under temporary directory")
+	}
+
+	if len(customMetadata) > 0 {
+		testJSON, err := json.MarshalIndent(customMetadata, "", "")
+		if err != nil {
+			return fmt.Errorf("error marshalling custom metadata: %v", err)
+		}
+		err = ioutil.WriteFile(filepath.Join(testArtifactDirectory, "custom-prow-metadata.json"), []byte(testJSON), os.FileMode(0644))
+		if err != nil {
+			return fmt.Errorf("unable to create custom metadata file: %v", err)
+		}
+	}
+
+	// Write a bunch of empty files that should be ignored
+	err = ioutil.WriteFile(filepath.Join(testArtifactDirectory, "a-ignore1.txt"), []byte(""), os.FileMode(0644))
+	err = ioutil.WriteFile(filepath.Join(testArtifactDirectory, "b-ignore1.txt"), []byte(""), os.FileMode(0644))
+	err = ioutil.WriteFile(filepath.Join(testArtifactDirectory, "d-ignore1.txt"), []byte(""), os.FileMode(0644))
+	err = ioutil.WriteFile(filepath.Join(testArtifactDirectory, "e-ignore1.txt"), []byte(""), os.FileMode(0644))
+	if err != nil {
+		return fmt.Errorf("one or more of the empty *ignore files failed to write: %v", err)
+	}
+
+	// Sleep for one second to guarantee we get a different modified time of the second write
+	time.Sleep(1 * time.Second)
+	err = o.writeMetadataJSON()
+	if err != nil {
+		return fmt.Errorf("error while writing metadata JSON: %v", err)
+	}
+
+	err = verifyMetadataFields(metadataFile, org, repo, extraOrg, extraRepo, prowJobID, namespace, customMetadata)
+	if err != nil {
+		return fmt.Errorf("error validating metadata: %v", err)
+	}
+
+	info, err = os.Stat(metadataFile)
+	if err != nil {
+		return fmt.Errorf("unable to stat metadata file second time: %v", err)
+	}
+
+	hasCustomMetadata := len(customMetadata) > 0
+	if hasCustomMetadata && !info.ModTime().After(originalModTime) {
+		return fmt.Errorf("file was not modified after the second function call when it should have due to the presence of custom metadata")
+	} else if !hasCustomMetadata && info.ModTime().After(originalModTime) {
+		return fmt.Errorf("file was modified after the second function call when it should not have due to the absence of custom metadata")
+	}
+
+	return nil
+}
+
+func createOptions(tempDir string, org string, repo string, extraOrg string, extraRepo string, prowJobID string, namespace string) *options {
```

Comment:
I'm not a fan of methods like these. Test should, when possible, contain all the logic that makes them tick in the `Test*` method itself. If you're worried about vertical space in your editor, take up some horizontal space instead:

```go
o := &options{
	artifactDir: tempDir,
	namespace: namespace,
	jobSpec: &api.JobSpec{JobSpec: downwardapi.JobSpec{
			ProwJobID: prowJobID,
			Refs: &prowapi.Refs{Org:  org, Repo: repo},
			ExtraRefs: []prowapi.Refs{{Org:  extraOrg, Repo: extraRepo}},
		},
	},
}
```